\documentclass{article}

\usepackage{fullpage}
\usepackage[doublespacing]{setspace}
\usepackage{amsmath, amssymb,amsfonts}
\usepackage[small]{caption}
\usepackage{graphicx}
\usepackage{xspace}
\usepackage{natbib}
\usepackage{datetime} %provides \currenttime command

\graphicspath{{./Figures/}}
\DeclareGraphicsExtensions{.pdf, .jpg, .png}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%Local Commands %%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\LogN}{\ensuremath{\text{LogN}}\xspace}

\newcommand{\EE}{\mathbb{E}} %use for expectation function E()
\newcommand{\setG}{\ensuremath{\mathbb{G}}\xspace}
\newcommand{\setP}{\ensuremath{\mathbb{P}}\xspace}
\newcommand{\qij}{\ensuremath{q_{i,j}}\xspace}
\newcommand{\qji}{\ensuremath{q_{i,j}}\xspace}
\newcommand{\pij}{\ensuremath{p_{i,j}}\xspace}
\newcommand{\muij}{\ensuremath{\mu_{i,j}}\xspace}
\newcommand{\fij}{\ensuremath{f_{i,j}}\xspace}
\newcommand{\Pii}{\ensuremath{\pi_{i}}\xspace}
\newcommand{\Pij}{\ensuremath{\pi_{j}}\xspace}
\newcommand{\Pivec}{\ensuremath{\Vec{\pi}}\xspace}
\newcommand{\Pivechat}{\ensuremath{\hat{\Pivec}}\xspace}
\newcommand{\Piihat}{\ensuremath{\hat{\pi}_i}\xspace}
\newcommand{\Pijhat}{\ensuremath{\hat{\pi}_j}\xspace}
\newcommand{\Wi}{\ensuremath{{W_i}}\xspace}
\newcommand{\Wj}{\ensuremath{{W_j}}\xspace}
\newcommand{\Ne}{\ensuremath{{N_e}}\xspace}
% \renewcommand{\ng}{\ensuremath{{n_g}}\xspace}
\newcommand{\DeltaAIC}{\ensuremath{\Delta\text{AIC}}\xspace}
\newcommand{\GTR}{GTR+$\Gamma$\xspace}

\newcommand{\jmax}{\ensuremath{{j_{\max}}}\xspace}
\newcommand{\kmax}{\ensuremath{{k_{\max}}}\xspace}
\newcommand{\cvec}{\ensuremath{\Vec{c}}\xspace}
\newcommand{\cveci}{\ensuremath{\cvec_i}\xspace}
\newcommand{\cvecj}{\ensuremath{\cvec_j}\xspace}

\newcommand{\acvec}{\ensuremath{a\left(\Vec{c}\right)}\xspace}
\newcommand{\acivec}{\ensuremath{a\left(\cveci\right)}\xspace}
\newcommand{\acvecg}{\ensuremath{a\left(\vec{c}_{i,g}\right)}\xspace}
\newcommand{\aip}{\ensuremath{a_{i,p}}\xspace}

\newcommand{\avec}{\ensuremath{\Vec{a}}\xspace}
\newcommand{\aveci}{\ensuremath{\Vec{a}_i}\xspace}
\newcommand{\aivec}{\aveci}
\newcommand{\aivecg}{\ensuremath{{\avec}_{i,g}}\xspace}

\newcommand{\avecj}{\ensuremath{\Vec{a}_j}\xspace}
\newcommand{\acvecj}{\ensuremath{a\left(\cvecj\right)}\xspace}


\newcommand{\aopt}{\ensuremath{a^*}\xspace}
\newcommand{\aoptip}{\ensuremath{\aopt_{i,p}}\xspace}
\newcommand{\aoptp}{\ensuremath{\aopt_p}\xspace}
\newcommand{\aoptpg}{\ensuremath{\aopt_{p,g}}\xspace}
\newcommand{\aoptvec}{\ensuremath{\Vec{a}^*}\xspace}
\newcommand{\aoptvecg}{\ensuremath{{{\aoptvec}_g}}\xspace}
\newcommand{\aobs}{\ensuremath{a_{\text{obs}}}\xspace}
\newcommand{\aobsvec}{\ensuremath{\Vec{a}_{\text{obs}}}\xspace}
\newcommand{\aobsvecg}{\ensuremath{{\avec}_{\text{obs},g}}\xspace}



\newcommand{\aj}{\ensuremath{a__j}\xspace}
\newcommand{\ajvec}{\ensuremath{\Vec{a}_{j}}\xspace}
\newcommand{\ajvecg}{\ensuremath{{\ajvec}_{,g}}\xspace}

\newcommand{\phig}{\ensuremath{\phi_{g}}\xspace}

\newcommand{\Cost}{\ensuremath{\text{Cost}}\xspace}
\newcommand{\Costaobsvec}{\ensuremath{\text{Cost}(\aobsvec)}\xspace}
\newcommand{\Costavec}{\ensuremath{\text{Cost}(\avec)}\xspace}
\newcommand{\Costaveci}{\ensuremath{\text{Cost}(\aveci)}\xspace}
\newcommand{\Costavecj}{\ensuremath{\text{Cost}(\avecj)}\xspace}

\newcommand{\Costcveci}{\ensuremath{\text{Cost}(\cveci)}\xspace}
\newcommand{\Costcvecj}{\ensuremath{\text{Cost}(\cvecj)}\xspace}

\newcommand{\Func}{\ensuremath{\text{Benefit}}\xspace}
\newcommand{\Funcaobsvec}{\ensuremath{\Func(\aobsvec|\aoptvec)}\xspace}
\newcommand{\Funcavec}{\ensuremath{\Func(\avec|\aoptvec)}\xspace}
\newcommand{\Funcaveci}{\ensuremath{\Func(\aveci|\aoptvec)}\xspace}
\newcommand{\Funcavecj}{\ensuremath{\Func(\avecj|\aoptvec)}\xspace}
\newcommand{\Funcaoptvec}{\ensuremath{\Func(\aoptvec)}\xspace}

\newcommand{\Funccvec}{\ensuremath{\Func(\cvec|\aoptvec)}\xspace}
\newcommand{\Funccveci}{\ensuremath{\Func(\cveci|\aoptvec)}\xspace}

\renewcommand{\ng}{\ensuremath{{n_g}}\xspace}


\title{SelAC Paper}
\date{}
\begin{document}
\maketitle

\section*{Introduction}
\begin{enumerate}
\item Phylogenetic methods play an important role in many fields of biology and medicine.
\item Essentially all phylogenetic approaches use a substitution matrix $Q = \left\{\qij\right\}$  to model evolution, where
  \begin{equation*}
   \qij  = \text{Substitution rate from state $i$ to $j$.}
  \end{equation*}
\item Most models, e.g.~F84, GTR, and GY94, use a `time reversible matrix' where $\pi_i \qij = \pi_j \qji $ for all $i \neq j$.
\item Time reversible matrices are very computationally efficient to employ.
\item Time reversible matrices (TRM) were initially derived under the assumption of neutrality but have been extended heuristically to describe non-neutral evolution.
\item However, non-neutral evolution is not a time reversible process, thus TRM models cannot accurately desribe evolutionary behavior when natural selection occurs.
\end{enumerate}



\paragraph*{Example of Nonsensical Behavior of Time Reversible Models:}
We illustrate the disconnect between time reversible models and non-neutral evolution we use a simplified version of the extremely popular Goldman and Yang (1994)\cite{GoldmanAndYang94} (GY94) codon level model.
In their model,
    \begin{align*}
      q_{i,j} &%= \text{Transition rate from $i \to j$} 
         = \begin{cases}
           0 & \text{$i$ and $j$ differ by more than one substitution}\\
           \Pijhat & \text{Synonymous (S) substitution} \\
           \omega \Pijhat & \text{Non-Synonymous (NS) substitution} \\
         \end{cases}
         \intertext{Where,}
         \omega &= \text{`Selection' term applied to all NS substitutions}\\
         \Pijhat &= \text{Equilibrium frequency of codon $i$}
       \end{align*}
When $\omega <1$ the GY94 model is purported to describe evolution under `purifying' selection where S substitutions are favored over NS substitutions.
However, the model has the following behavior
    \begin{enumerate}
    \item If $i$ is the current state, GY94 implies selection favoring $i$. \label{pureone}
    \item However, if NS substitution occurs, \ref{pureone} still applies and selection now favoring new state $j$!
    \end{enumerate}
Thus, the behavior of GY94 is actually not consistent with a constant selective environment, but instead is consistent with a system where the directionality of natural selection and a NS substitution occurs simultaneously.
Similar inconsistencies occur when $\omega < 1$.


\section*{Methods}
\subsection*{Allele Substitution Model}
\subsubsection*{Defining Protein Synthesis Cost-Benefit Function $\eta$: }
Because our model assumes that natural selection favors genotypes that are able to meet their metabolic requirements more efficiently than their competitors, our framework centers on the cost-benefit function of a gene $g$, $\eta_g$, and the organisms average target production rate of the functionality provided by gene $g$, $\phig$.
This is because the average amount of energy an organism spends to met its target functionality for a gene $g$ is $\eta_g \times \phig$.
 
\paragraph*{Defining the Cost Function}
Generally speaking, protein synthesis involves both direct and indirect assembly costs.
Direct costs consist of the high energy phosphate bonds in ATP or GTP's used to assemble the ribosome on the mRNA, charge tRNA's for elongation, move the ribosome forward along the transcript, and terminate protein synthesis.
Indirect costs are many and consist of the cost of amino acid synthesis as well as synthesis of the protein assembly infrastructure such as ribosomes, aminoacyl-tRNA synthetases, tRNAs, mRNAs, etc.
Direct synthesis costs are the same for all proteins of the same length.
For simplicity, in this study we ignore any indirect costs of protein synthesis that vary between genotypes.
As a result, 
\begin{align}
\label{eq:defineCost}
  \Costcveci  &= \text{Energetic cost of protein synthesis.}\\
  &= C_1 + C_2 n
\end{align}
where, $C_1$ and $C_2$ represent the direct and indirect costs in ATPs of ribosome initiation and peptide elongation, respectively.
When sequence specific costs, such as ribosome pausing times, are included with sequence specific benefits, the probability of a mutant allele fixing is no longer independent of the rest of the sequence.
As a result, our site independent assumption is violated and the fitting of our model becomes much more complex.
For simplicity, in this study we only consider the direct costs of protein assembly and, thus, $C_1 = C_2 = 4 \text{ATP}$.
 
\paragraph*{Defining the Benefit Function: }
In order to link genotype to protein function, we define a benefit function which measures the functionality of the peptide encoded by \cveci, i.e. $a(\cveci) = \aveci$ relative to the the optimal sequence $\aoptvec$.
By definition, we set $\Funcaoptvec = 1$ and assume $\Funcaveci < 1$ for all $\aveci$ other than the optimal sequence.
How protein functionality declines with deviation from \aoptvec, is an overwhelmingly complex problem and likely varies between different categories of proteins.
Instead of claiming to accurately model this relationship between genotype and protein function, we will fit a Taylor Series expansion to our data in order to approximate its general behavior.
We will also assume a form that results in independent evolution between sites within a gene.
Alternative forms of \Funcaveci can, of course, be explored by other researchers.

To begin, we assume that each amino acid makes a similar contribution to protein function (an assumption that can be relaxed) and that this contribution declines as an inverse function of physiochemical distance.
More specifically, we assume 
\begin{equation}
\Funcaveci = \left(\frac{1}{\ng} \sum_p^\ng{} f\left(d\left(\aip, \aoptp\right)\right)\right)^{-1}
\end{equation}
where $\ng$ is the length of the protein, $d(\aip, \aoptp)$ is the physiochemical distance between the amino acid encoded in gene $i$ for position $p$ and $\aoptp$ is the optimal amino acid for that position of the protein, and $1/f(d)$ describes how the contribution of amino acid to protein function declines with $d$.

How $f(d)$ changes with $d$ is unknown, as a result we use a combination of a Taylor Series expansion and random effect to describe the relationship between $f$ and $d$.
Given our assumption that \Funcaoptvec = 1 and noting that $d$ has its own free parameters $\alpha$ and $\beta$, we define $f(d)$ as,
\begin{align}
  \label{eq:fSeriesDef}
  f(d) &= 1 + g \sum_{k=1}^\kmax \frac{1}{k!}\frac{d f^k}{d^k d} d^k + O(d^{\kmax+1})\\
  & = 1 + g \sum_{k=1}^\kmax A_k d^k + O(d^{\kmax+1})\\
\end{align}
where we define $A_k = \frac{1}{k!}\frac{d f^k}{d^k d}$ in order to emphasize the polynomial nature of our approximation to $f(d)$ and use $g$ to represent a random effect.
Here we assume $g \sim \text{Gamma}\left(\alpha_g, \beta_g = 1/\alpha_g\right)$ in order to ensure $\EE(g) = 1$, but other functions could be used.

Because $\phi$ and $A_1$ always co-occur, we cannot identify them separately from one another; as a result, we set $A_1 = 1$ and recognize that our estimates of $\phi$ are scaled relative to this term.
Using the results from \citet{Liang07} and \citet{Elphinstone85}, we can ensure that $f(d)$ is a monotonic, increasing function of $d$ by fitting our model using a transformation of variables $\alpha$ and $\beta$ and by restricting \kmax to multiples of 2. (Note that because $d > 0$, $f(d)$ is monotonic and increasing when $\kmax=1$.)
%Since we are assuming $A_0$ and $A_1 = 1$, when using the recurrence relation in Lang 2007, we set $\lambda = 1$ and $\alpha_1 = -1/2$.


\paragraph*{Defining Physiochemical Distances between Amino Acids :}
Assuming that functionality declines with an amino acid $a_i$'s physiochemical distance from the optimum amino acid \aopt at each site  provides a biologically defensible way of linking comparing genotypes that requires relatively few free parameters.
In addition, our approach naturally lends itself to model selection since we can compare the quality of our model fits using different mixtures of physiochemical properties.
Following \citet{Grantham74}, we focus on using composition $c$, polarity $p$, and molecular volume $v$ of each amino acid's side chain residue to define our distance function, but emphasize that other properties could be used.
We use the euclidian distance between residue properties where each property $c$, $p$, and $v$ has its own weighting term, $\alpha$, $\beta$, $\gamma$, respectively, [NOTE: WE MAY WANT TO USE $\alpha_c, \alpha_p, \ldots $ INSTEAD].
Because of similar identifiability issues we have with $A_1$ and $\phi$, we set $\gamma = 1$ and recognize that our our estimates of $\alpha$ and $\beta$ are scaled relative to $\gamma$.
More specifically,
\begin{equation*}
  d(a_i, \aopt) = \sqrt{\alpha \left(c\left(a_i\right) - c\left(\aopt\right)\right)^2 + \beta \left(p\left(a_i\right) - p\left(\aopt\right)\right)^2 +  \gamma \left(v\left(a_i\right) - v\left(\aopt\right)\right)^2}.
\end{equation*}



\subsubsection*{Linking Cost of Protein Synthesis to Allele Fixation}
In order to link the protein synthesis cost-benefit function $\eta$ of an allele with its fixation probability, we must make a number of assumptions.
First, we assume that each protein encoded within a genome carries out some beneficial function and that the organism needs that functionality to be produced at a target average rate $\phi$.
By definition, the optimal amino acid sequence for a given gene, \aoptvec, produces one unit of functionality.
Second, we assume that protein expression is regulated by the organism to ensure that functionality is produced at rate $\phi$.
As a result, the average protein production rate of a gene, $\psi$, is equal to $\phi/\Func(\avec)$ and the total energy flux allocated towards meeting the target functionality of a particular gene is $\eta(\cvec) \phi$. 
Third, we assume that every additional ATP spent per unit time to meet the organism's target function production rate $\phi$ leads to some slight proportional incremental decrease in fitness $W$.
This assumption, in turn, implies 
\begin{align}
  W_i\left(\cvec\right) &\propto \exp\left[- q \eta(\cveci) \phi\right].
\end{align}
where $q$ describes the decline in fitness with every ATP wasted per unit time $\phi$ and $\psi$ are measured in.
Correspondingly, the ratio of fitness between two genotypes is,
\begin{align*}
  W_i/W_j &=  \exp\left[- q \eta(\cveci) \phi\right]/\exp\left[- q \eta(\cvecj) \phi\right]\\
  &=  \exp\left[- q \left(\eta(\cveci)- \eta(\cvecj)\right) \phi\right]\\
\end{align*}
Given our assumptions about the \Cost and \Func functions above, this ratio simplifies to
\begin{align}
  W_i/W_j &\approx \left[- q \left(C_1 + C_2 n\right) \frac{1}{n}\left(\sum_{p \in \setP} \sum_{k=1}^\kmax A_k \left(d\left(\aip,\aoptp\right)^k - d\left(ajp,\aoptp\right)\right)\right) \phi \right]
%      &=  \exp\left[- q \left(C_1/n + C_2\right)\left(\sum_{p \in \setP} \sum_{k=1}^\kmax A_k \left(d\right(\aip,\aoptp\right)^k - d\left(ajp,\aoptp\right)^k\right) \phi\right]
\end{align}
where \setP represents the codon positions in which \cveci and \cvecj differ.
Fourth, we make a weak mutation assumption, such that alleles can differ at only one position at any given time, i.e.~$|\setP| = 1$, and that the population is evolving according to a Fisher-Wright model.
As a result, the probability a new mutant $j$ introduced via mutation into a resident population $i$ with effective size \Ne will go to fixation is,
\begin{align*}
  u_{i,j} &=  \frac{1 - \left(W_i/W_j\right)^b}{1 - \left(W_i/W_j\right)^\Ne}\\
   &\approx \frac{1-\exp\left[- b q \left(\eta(\cveci)- \eta(\cvecj)\right) \phi\right]}{1-\exp\left[- q \left(\eta(\cveci)- \eta(\cvecj)\right) \phi 2\Ne\right]}\\
   &= \frac{1- \exp\left[- b q \left(C_1/n + C_2\right)\left(\sum_{k=1}^\kmax A_k \left(d\right(\aip,\aoptp\right)^k - d\left(ajp,\aoptp\right)^k\right)\left) \phi\right]}{1-\exp\left[- q \left(C_1/n + C_2\right)\left(\sum_{k=1}^\kmax A_k \left(d\right(\aip,\aoptp\right)^k - d\left(ajp,\aoptp\right)^k\right)\left) \phi 2 \Ne\right]},
\end{align*}
where $b=1$ for a diploid population and $2$ for a haploid population \cite{Iwasa88,BergAndLassig03,SellaAndHirsh05}.
Finally, assuming a constant mutation between alleles $i$ and $j$, $\mu_{i,j}$, the transition rate from allele $i$ to $j$ can be modeled as,
\begin{align*}
  q_{i,j} = \begin{cases} 2 \mu_{i,j}/b & \text{if alleles $i$ and $j$ differ by one nucleotide}\\
    0 & \text{else}
  \end{cases}
\end{align*}




\section*{Discussion}

\begin{itemize}
\item Note that our definition of $\phi$ and our scaling of functionality differ slightly from our previous work \citep{Gilchrist07,GilchristEtAl09,ShahAndGilchrist11,GilchristEtAl15a}.
In our previous work, we were concerned with how changes in synonymous codons affected error rates and synthesis costs and, as a result, defined functionality relative to an error free protein, rather than an optimal one, and conflated $\phi$ and $\psi$.
\item Our approach requires relatively few parameter. 
  \begin{enumerate}
  \item Distance function $d(a_i, \aopt)$: If $n_d$ is the number of physiochemical properties examined, the number of parameters estimated is $n_d - 1$
  \item Benefit function $\Func$: If $n_A$ is the order of our Taylor Series approximation, the number of parameters is $n_A-1$.
  \item Gene expression $\phi$: One $\phi$ for each gene analyzed.
  \item Mutation bias: Depends on the model used it is either equal to the number of parameters in the model $n_\mu$ or $n_\mu-1$.
  \end{enumerate}
\item Statistical Physics model allows decoupling of \Ne, $\mu$, and strength selection.
\end{itemize}
\bibliographystyle{./am.nat}
\bibliography{./mike}

\subsubsection*{Notes for Jeremy}

\section*{Log}
\begin{itemize}
\item ``Defining the Benefit Function'' section began by mikeg on 7/23/15.
\item Methods expanded to include ``Defining Benefit Function'' on ?
\item {Linking Genotype Energetics to Fitness and Fixation} subsection added to Model.
\item Compiled on \today\xspace at \currenttime. 
\end{itemize}

\end{document}
